controllers/cluster_controller.go

```diff
@@ -592,7 +592,23 @@ func (r *FoundationDBClusterReconciler) getStatusFromClusterOrDummyStatus(logger
 		return status, nil
 	}
 
+	// When we reached this part of the code the above GetStatus() called failed for some reason.
 	if cluster.Status.Configured {
+		// If the cluster is currently under a version incompatible upgrade, we try to assume the current version based
+		// on the coordinator reachability. If all (or the majority) of coordinators are reachable with a specific version
+		// of fdbcli we can assume that the cluster is running with that version and update the cluster.Status.RunningVersion.
```

Comment:
or is it more that after we've done a bounce, if the coordinators are up then we know the cluster is running that version?

Reply:
This case is only here to help the operator to detect the running version when the `machine-readable` status is not available, e.g. the above `GetStatus` returned an error. If the status is returned, the operator will detect the running version based on the running processes. So the intent here is to help the operator detect the running version based on the reporting coordinators (so those that are reachable in a specific version), to unblock the operator and let it move forward. The reporting coordinators will be either reachable in a quorum of the old version, the new version or none, but all those three states are distinct, e.g. if a quorum is reachable it will be either in the old or the new version but cannot be both.